components/automate-gateway/gateway/middleware/authv2/authv2.go

```diff
@@ -83,13 +83,44 @@ func (c *client) Handle(ctx context.Context, subjects []string, projectsToFilter
 	return auth_context.NewContext(ctx, subjects, projects, resource, action, middleware.AuthV2.String()), nil
 }
 
+type resp struct {
+	ctx        context.Context
+	authorized bool
+}
+
+
+func (r *resp) Ctx() context.Context {
+	return r.ctx
+}
+
+func (r *resp) GetAuthorized() bool {
+	return r.authorized
+}
+
 func (c *client) IsAuthorized(ctx context.Context, subjects []string, resource, action string,
-) (middleware.AuthorizationResponse, error) {
-	return c.client.IsAuthorized(ctx, &authz.IsAuthorizedReq{
-		Subjects: subjects,
-		Resource: resource,
-		Action:   action,
+	projectsToFilter []string) (middleware.AuthorizationResponse, error) {
+	log := ctxlogrus.Extract(ctx)
+	filteredResp, err := c.client.ProjectsAuthorized(ctx, &authz.ProjectsAuthorizedReq{
+		Subjects:       subjects,
+		Resource:       resource,
+		Action:         action,
+		ProjectsFilter: projectsToFilter,
 	})
+	if err != nil {
+		if status.Convert(err).Code() == codes.FailedPrecondition {
+			return nil, err
+		}
+		log.WithError(err).Error("error authorizing request")
+		return nil, status.Errorf(codes.PermissionDenied,
```

Comment:
Should this message/error also include projects?

Reply:
As this is the gatekeeper for both v2 and v2.1, and it does not know which for any given call, we cannot appropriately include projects.